### PR TITLE
Make warning as error for cgo

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Turn clang warnings into errors.

### Additional Notes

https://github.com/DataDog/datadog-agent/blob/7.22.1/pkg/collector/python/test_loader.go#L73 generates a warning as `version` is `const char*` and `free` expects a `char*`. As the function was not used, it is removed.

### Describe your test plan

As only the declarations are updated, nothing need to be tested.

### CODE REVIEW

**Review by hush-hush — APPROVED** [review]

**Comment by ogaca-dd** [comment]
@hush-hush : I did a small change on this PR that need a review.

File: tasks/utils.py [hunk 1]
```diff
@@ -108,9 +108,12 @@ def get_build_flags(
     if rtloader_lib:
         env['DYLD_LIBRARY_PATH'] = os.environ.get('DYLD_LIBRARY_PATH', '') + ":{}".format(':'.join(rtloader_lib))  # OSX
         env['LD_LIBRARY_PATH'] = os.environ.get('LD_LIBRARY_PATH', '') + ":{}".format(':'.join(rtloader_lib))  # linux
-        env['CGO_LDFLAGS'] = os.environ.get('CGO_LDFLAGS', '') + " -L{}".format(' -L '.join(rtloader_lib))
-    env['CGO_CFLAGS'] = os.environ.get('CGO_CFLAGS', '') + " -w -I{} -I{}".format(
-        rtloader_headers, rtloader_common_headers
+        env['CGO_LDFLAGS'] = os.environ.get('CGO_LDFLAGS', '') + " -L{}".format(' -L '.join(rtloader_lib))    
+    warning_as_error = '-Werror'
+    if sys.platform == 'win32':
+        warning_as_error = '' # Disable on Windows because of redefined macro errors.
```
Comment (albertvaka) [inline tasks/utils.py:114]:
```suggestion
        warning_as_error = '' # FIXME: Disable on Windows because of redefined macro errors.
```

**Review by albertvaka — APPROVED** [review]
LGTM, but later we should also enable this on Windows.

**Comment by ogaca-dd** [comment]
commits from "test"  (f2f5528) to "fix warning" (226e75a") can be ignored.
Only the last commits are relevant (starting `Remove the defines ERROR_EVT_* as GNU 7.4.0 is used.` (ee6d22f))

**Comment by ogaca-dd** [comment]
Wrong merge reopening this PR.

**Pull request closed without merging**